game-app/game-headed/src/main/java/games/strategy/engine/framework/startup/ui/ServerSetupPanel.java

```diff
@@ -380,50 +422,20 @@ public void actionPerformed(final ActionEvent e) {
               type.setSelectedItem(PlayerTypes.WEAK_AI);
             }
             setWidgetActivation();
-          }
-        };
-
-    PlayerRow(
-        final String playerName,
-        final Map<String, String> reloadSelections,
-        final Collection<String> playerAlliances,
-        final PlayerTypes playerTypesProvider) {
-      nameLabel = new JLabel(playerName);
-      playerLabel = new JLabel(model.getMessenger().getLocalNode().getName());
-      localCheckBox = new JCheckBox();
-      localCheckBox.addActionListener(localPlayerActionListener);
-      localCheckBox.setSelected(true);
-      enabledCheckBox = new JCheckBox();
-      enabledCheckBox.addActionListener(disablePlayerActionListener);
+          });
       // this gets updated later
       enabledCheckBox.setSelected(true);
-      final String[] playerTypes = playerTypesProvider.getAvailablePlayerLabels();
-      type = new JComboBox<>(playerTypes);
-      String previousSelection = reloadSelections.get(playerName);
-      if (previousSelection.equalsIgnoreCase("Client")) {
-        previousSelection = playerTypes[0];
-      }
-      if (!previousSelection.equals("no_one") && List.of(playerTypes).contains(previousSelection)) {
-        type.setSelectedItem(previousSelection);
-        model.setLocalPlayerType(
-            nameLabel.getText(), playerTypesProvider.fromLabel((String) type.getSelectedItem()));
-      } else if (playerName.startsWith("Neutral") || playerName.startsWith("AI")) {
-        // the 4th in the list should be Pro AI (Hard AI)
-        type.setSelectedItem(PlayerTypes.PRO_AI.getLabel());
-        model.setLocalPlayerType(nameLabel.getText(), PlayerTypes.PRO_AI);
-      }
       if (playerAlliances.contains(playerName)) {
         alliance = new JButton();
         alliance.setVisible(false);
       } else {
         alliance = new JButton(playerAlliances.toString());
-        alliance.setToolTipText("Click to play " + playerAlliances.toString());
+        alliance.setToolTipText("Click to play " + playerAlliances);
       }
       type.addActionListener(
           e ->
               model.setLocalPlayerType(
-                  nameLabel.getText(),
-                  playerTypesProvider.fromLabel((String) type.getSelectedItem())));
```

Comment:
is the fromLabel method still used after this?

Reply:
Unfortunately, yes - in PlayerSelectorRow - which is the single-player version of this code. It would be good to unify them eventually...